packages/world/src/World.sol

```diff
@@ -51,6 +51,9 @@ contract World is StoreData, IWorldKernel {
 
   /**
    * @dev Prevents the World contract from calling itself.
+   * The world is not able to call itself; all operations to internal tables happen as internal library calls, and all calls to root system happen as a delegatecall to the system.
+   * However, since this is an important invariant, we make it explicit by reverting if `msg.sender` is `address(this)` in all `World` methods.
+   * If it was possible to make the `World` call itself, it would be possible to access internal tables that only the `World` should have access to.
```

Comment:
this feels a bit incomplete, like we should either expand on how/why or just remove

> If the `World` is able to call itself via `delegatecall` from a system, this would be a backdoor where that system would have "root" access to things like internal tables.

(maybe could workshop this exact sentence a bit more, but just providing a bit more context)